src/optimum/rbln/transformers/modeling_attention_utils.py

```diff
@@ -115,128 +115,261 @@ def validate_sliding_window(rbln_config: RBLNDecoderOnlyModelForCausalLMConfig):
+        alloc_memory_by_key.pop("PortRecur", None)  # Old compiler's kv-cache Key
+        alloc_memory_by_key.pop("DramTensor", None)  # kv-cache
+        used_memory = sum(alloc_memory_by_key.values())
+
+        remaining_dram = available_dram - used_memory
+
+        if remaining_dram <= 0:
+            logger.warning(
+                "Insufficient available DRAM after accounting for kernel memory and buffer. "
+                "Model cannot allocate any KV cache blocks."
+            )
+
+        estimated_num_blocks = cls._estimate_num_blocks(
+            remaining_dram, model_config=model_config, rbln_config=rbln_config
+        )
+
+        return estimated_num_blocks
+
+    @classmethod
+    def _estimate_num_blocks(
+        cls, available_dram: int, model_config: "PretrainedConfig", rbln_config: RBLNDecoderOnlyModelForCausalLMConfig
+    ) -> int:
+        """
+        Estimate the maximum number of KV cache blocks that can be allocated.
+
+        if all of the layers are full attention, the dram_per_block can be calculated simply as follows:
+            num_blocks = available_dram // dram_per_block
+
+        However, if the model contains a mix of full attention and sliding window attention layers,
+        we need to consider the memory occupied by the sliding window attention layers first,
+        since their memory usage is constant regardless of the number of blocks.
+            num_blocks = (available_dram - swa_kv_nbytes) // dram_per_block
+
+        """
+
+        def get_dram_per_block(seq_len: int, num_key_value_heads: int, tensor_parallel_size: int) -> int:
+            nbytes_per_param = 2  # Assuming kv-cache is always not quantized
+            dram_per_block = (
+                seq_len
+                * align(head_dim, 64)
+                * math.ceil(num_key_value_heads / tensor_parallel_size)
+                * nbytes_per_param
                 * tensor_parallel_size
+                * 2
+            )  # *2 for key and value
+
+            return dram_per_block
+
+        num_attention_heads = getattr(model_config, "n_head", None) or getattr(model_config, "num_attention_heads")
+        head_dim = getattr(model_config, "head_dim", None) or model_config.hidden_size // num_attention_heads
+        num_hidden_layers = getattr(model_config, "n_layer", None) or getattr(model_config, "num_hidden_layers")
+        num_key_value_heads = getattr(model_config, "num_key_value_heads", None) or num_attention_heads
+        tensor_parallel_size = rbln_config.tensor_parallel_size or 1
+
+        # Consider layer types if available
+        # If layer types are not found, assume all layers are full attention
+        layer_types = getattr(model_config, "layer_types", None)
+        if layer_types:
+            layer_types_dict = Counter(layer_types)
+            num_full_attention = layer_types_dict.pop("full_attention", 0)
+            num_sliding_window_attention = layer_types_dict.pop("sliding_window_attention", 0)
```

Comment:
```suggestion
            num_sliding_window_attention = layer_types_dict.pop("sliding_attention", 0)
```
The name of sliding window attention in layer types is `sliding_attention`.
- https://github.com/huggingface/transformers/blob/8cb5963cc22174954e7dca2c0a3320b7dc2f4edc/src/transformers/configuration_utils.py#L1366
- https://huggingface.co/LGAI-EXAONE/EXAONE-4.0-32B/blob/main/config.json